### PR TITLE
Fix/optimized loading of assamblys with nopsp

### PR DESCRIPTION
**Pull request by ferryvandegraaf**
# Describe your changes

When PSP is noPsp it is not necessary to load the assembly via refactoring, because noPSP is a class within the GCL. Optimized this for better performance.

## Type of change

Please check only ONE option.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How was this tested?

Tested by stepping through the lines and placed an order with NoPSP, this order was placed succesfully

# Checklist before requesting a review
- [x] I have reviewed and tested my changes
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I selected `develop` as the base branch and not `main`, or the pull request is a hotfix that needs to be done directly on `main`
- [x] I double checked all my changes and they contain no temporary test code, no code that is commented out and no changes that are not part of this branch
- [ ] I added new unit tests for my changes if applicable

# Related pull requests

Add any open pull requests from other projects that are related to this pull request that should be merged along with this one. If there are none, you can simply say "none" or "N/A", or just leave this section empty.

# Link to Asana ticket

Add a link to the Asana ticket. Note that **_ONLY_** the URL should be added, not the name of the ticket!

### CODE REVIEW

**Comment by HansHappyHorizon** [comment]
Please fill out the entire form instead of only checking some boxes (summary of changes and explanation of how the changes were tested).

File: GeeksCoreLibrary/Components/OrderProcess/Services/OrderProcessesService.cs [hunk 1]
```diff
@@ -884,7 +884,8 @@ public async Task<PaymentRequestResult> HandlePaymentRequestAsync(IOrderProcesse
                 var basketSettings = await shoppingBasketsService.GetSettingsAsync();
                 foreach (var (main, lines) in shoppingBaskets)
                 {
-                    if (paymentMethodSettings.PaymentServiceProvider.Type != PaymentServiceProviders.NoPsp && basketToConceptOrderMethod == OrderProcessBasketToConceptOrderMethods.Convert)
+                    var copyBasketToConceptOrderOnPspPayment  = (await objectsService.FindSystemObjectByDomainNameAsync("psp_copy_basket_to_concept_order","1")) == "1";
+                    if (paymentMethodSettings.PaymentServiceProvider.Type != PaymentServiceProviders.NoPsp && basketToConceptOrderMethod == OrderProcessBasketToConceptOrderMethods.Convert && copyBasketToConceptOrderOnPspPayment)
```
Comment (GilianJuice) [inline GeeksCoreLibrary/Components/OrderProcess/Services/OrderProcessesService.cs:888]: This code is from a pull request from Rinus (https://github.com/happy-geeks/geeks-core-library/pull/548), why is it also in yours? I denied that PR from Rinus, so please remove that code here as well.

File: GeeksCoreLibrary/Core/Extensions/DataRowExtensions.cs [hunk 2]
```diff
@@ -129,16 +133,23 @@ public static JObject ToJsonObject(this DataRow dataRow, string requiredColumnNa
...
-                        value = value.EncryptWithAes(encryptionKey, withDate);
+                        value = value.EncryptWithAes(encryptionKey, withDate, useSlowerButMoreSecureMethod:saferMethod);
+                    }
+                    if (saferMethod)
+                    {
+                        result.TryAdd(TrimPrefix(columnName[..^(withDate ? SaferEncryptWithDateSuffix : SaferEncryptSuffix).Length], requiredColumnNamePrefix), value.UrlEncode());
+                    }
+                    else
+                    {
+                        result.TryAdd(TrimPrefix(columnName[..^(withDate ? NormalEncryptWithDateSuffix : NormalEncryptSuffix).Length], requiredColumnNamePrefix), value.UrlEncode());
```
Comment (GilianJuice) [inline GeeksCoreLibrary/Core/Extensions/DataRowExtensions.cs:151]: What is all of this? This seems to have nothing to do with loading assembly for NoPSP.

**Review by GilianJuice — CHANGES_REQUESTED** [review]
I can't approve this. This PR contains a lot of changed that have nothing to do with what it said in the description.

To be honest, I also don't see much point in the change of not loading the NoPSP assembly. I don't see how this change makes a lot of impact, it might improve the speed by 1 millisecond, but it causes the problem that we can't add settings to NoPSP in the future. Why exactly did you make this change? Did you notice bad performance?

If it does actually make a noticable improvement, I can accept it, but not with all these other changes that have nothing to do with it.

**Pull request closed without merging**